recipes-extended/xen/files/0001-python-pygrub-pass-DISTUTILS-xen.4.12.patch

```diff
@@ -21,27 +21,24 @@ Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
 
 --- a/tools/pygrub/Makefile
 +++ b/tools/pygrub/Makefile
-@@ -10,7 +10,7 @@ INSTALL_LOG = build/installed_files.txt
+@@ -13,14 +13,15 @@ setup.py = CC="$(CC)" CFLAGS="$(PY_CFLAG
  all: build
  .PHONY: build
  build:
--	CC="$(CC)" CFLAGS="$(PY_CFLAGS)" LDSHARED="$(CC)" LDFLAGS="$(PY_LDFLAGS)" $(PYTHON) setup.py build
-+	CC="$(CC)" CFLAGS="$(PY_CFLAGS)" LDSHARED="$(CC)" LDFLAGS="$(PY_LDFLAGS)" $(PYTHON) setup.py build $(DISTUTILS_BUILD_ARGS)
+-	$(setup.py) build
++	$(setup.py) build $(DISTUTILS_BUILD_ARGS)
  
  .PHONY: install
  install: all
-@@ -18,7 +18,10 @@ install: all
- 	CC="$(CC)" CFLAGS="$(PY_CFLAGS)" LDSHARED="$(CC)" \
- 		LDFLAGS="$(PY_LDFLAGS)" $(PYTHON) setup.py install \
- 		--record $(INSTALL_LOG) $(PYTHON_PREFIX_ARG) \
--		 --root="$(DESTDIR)" --install-scripts=$(LIBEXEC_BIN) --force
-+		 --root="$(DESTDIR)" --install-scripts=$(LIBEXEC_BIN) --force \
-+		 $(DISTUTILS_INSTALL_ARGS)
+ 	$(INSTALL_DIR) $(DESTDIR)/$(bindir)
+ 	$(INSTALL_DIR) $(DESTDIR)/$(LIBEXEC_BIN)
+ 	$(setup.py) install --record $(INSTALL_LOG) $(PYTHON_PREFIX_ARG) \
+-		--root="$(DESTDIR)" --force
++		--root="$(DESTDIR)" --force $(DISTUTILS_INSTALL_ARGS)
 +	rm -f $(DESTDIR)/$(LIBEXEC_BIN)/pygrub
```

Comment:
I dropped this `rm -f $(DESTDIR)/$(LIBEXEC_BIN)/pygrub`, but I didn't write down why.  I think it just isn't necessary since it should be overwritten in the next line?  (Not saying you should change - just noting.)

Reply:
Might be something in the `install-wrap` stuff xen does that might choose not to overwrite it.  tools/python/install-wrap   does a `mv -f` so it's forcing it.  Idk, I don't see a substantial downside to cleaning the target destination before installing the pygrub target.  You're right it's probably not necessary though.